crates/nu-cli/src/completions/operator_completions.rs

```diff
@@ -0,0 +1,129 @@
+    fn fetch(
+        &mut self,
+        working_set: &StateWorkingSet,
+        _stack: &Stack,
+        _prefix: Vec<u8>,
+        span: Span,
+        offset: usize,
+        _pos: usize,
+        _options: &CompletionOptions,
+    ) -> Vec<SemanticSuggestion> {
+        //Check if int, float, or string
+        let partial = std::str::from_utf8(working_set.get_span_contents(span)).unwrap_or("");
+        let possible_operations = match &self.previous_expr_shape {
+            FlatShape::Int => vec![
+                ("+", "Plus / Addition"),
+                ("-", "Minus / Subtraction"),
+                ("*", "Multiply"),
+                ("/", "Divide"),
+                ("==", "Equal"),
+                ("!=", "Not Equal"),
+                ("//", "Floor Division"),
+                ("<", "Less Than"),
+                (">", "Greater Than"),
+                ("<=", "Less Than or Equal to"),
+                (">=", "Greater Than or Equal to"),
+                ("mod", "Modulo"),
+                ("**", "Pow"),
+                ("bit-or", "bitwise or"),
+                ("bit-xor", "bitwise exclusive or"),
+                ("bit-and", "bitwise and"),
+                ("bit-shl", "bitwise shift left"),
+                ("bit-shr", "bitwise shift right"),
+            ],
+            FlatShape::String => vec![
+                ("=~", "Regex Match / Contains"),
+                ("!~", "Not Regex Match / Not Contains"),
+                ("in", "In / Contains (doesn't use regex)"),
+                (
+                    "++",
+                    "Appends two lists, a list and a value, two strings, or two binary values",
+                ),
+                ("not-in", "Not In / Not Contains (doesn't use regex"),
+                ("starts-with", "Starts With"),
+                ("ends-with", "Ends With"),
+            ],
+            FlatShape::Float => vec![
+                ("+", "Plus / Addition"),
+                ("-", "Minus / Subtraction"),
+                ("*", "Multiply"),
+                ("/", "Divide"),
+                ("=", "Equal"),
+                ("!=", "Not Equal"),
+                ("//", "Floor Division"),
+                ("<", "Less Than"),
+                (">", "Greater Than"),
+                ("<=", "Less Than or Equal to"),
+                (">=", "Greater Than or Equal to"),
+                ("mod", "Modulo"),
+                ("**", "Pow"),
+            ],
+            FlatShape::Bool => vec![
+                ("and", "Checks if both values are true."),
+                ("or", "Checks if either value is true."),
+                ("xor", "Checks if one value is true and the other is false."),
+                ("not", "Negates a value or expression."),
+            ],
+            FlatShape::List => vec![(
+                "++",
+                "Appends two lists, a list and a value, two strings, or two binary values",
+            )],
+            FlatShape::Variable(_) => vec![
```

Comment:
I seemingly can't get this one to work. If non-trivial to get working we can also drop this part.

For correctness there are also limitations, that reassignment is only permitted for mutable variables declared by `mut` or part of `$env` so at least the documentation here should reflect that or even more correct the suggestion should be limited to those.

Reply:
This is probably because all the variable completions are returning VariableCompletion's suggestions. I'll try to refactor the helper so multiple completers can add suggestions before returning if it isn't super problematic.  https://github.com/nushell/nushell/blob/8200831b07ab87ab23c4a56a8a08251d0332b114/crates/nu-cli/src/completions/completer.rs#L177-L189

Reply:
Okay, I think variable completions should work as expected now, with a check for their mutability and type.